src/services/messengerSync/services/messengerTokenService.js

```diff
@@ -0,0 +1,96 @@
+const {	GeneralError, BadRequest } = require('@feathersjs/errors');
+const { authenticate } = require('@feathersjs/authentication');
+const { disallow } = require('feathers-hooks-common');
+const { Configuration } = require('@schul-cloud/commons');
+const request = require('request-promise-native');
+const hmacSHA512 = require('crypto-js/hmac-sha512');
+const { createChannel } = require('../../../utils/rabbitmq');
+
+
+function obtainAccessToken(userId, homeserverApiUri, secret) {
+	const loginApiUrl = `${homeserverApiUri}/_matrix/client/r0/login`;
+
+	const password = hmacSHA512(userId, secret).toString();
+
+	const payload = {
+		type: 'm.login.password',
+		user: userId,
+		password,
+	};
+
+	const options = {
+		uri: loginApiUrl,
+		method: 'POST',
+		body: payload,
+		json: true,
+	};
+	return request(options)
+		.then((response) => {
+			const session = {
+				userId,
+				homeserverUrl: homeserverApiUri,
+				accessToken: response.access_token,
+			};
+			return session;
+		});
+}
+
+class MessengerTokenService {
+	constructor(options) {
+		this.options = options || {};
+	}
+
+	async setup(app) {
+		this.app = app;
+		this.channel = await createChannel();
+	}
+
+	/**
+	 * syncs the rooms of all users on the school.
+	 * @param {Object} data should contain a schoolId field
+	 * @param {Object} params feathers params object.
+	 */
+	async create(data, params) {
+		if (!Configuration.get('FEATURE_MATRIX_MESSENGER_ENABLED')) {
+			throw new GeneralError('messenger not supported on this instance');
+		}
```

Comment:
Ich würde den Service gar nicht erst registrieren, wenn das Feature nicht aktiv ist

Reply:
hatte ich am anfang auch [berlegt, aber so ist die Fehlermeldung besser wenn doch jemand entweder den Service ruft obwohl er das nciht sollte, oder das flag in der config vergessen wurde.

Reply:
Ich glaube das könnte der Service ja beim Start sicherstellen. Wenn ich mir den Service alleinstehend vorstelle, dann muss er ja auch beim Startup alle configs lesen und eventuell mit Fehler abstürzen; hier registriert sich der Service trotzdem und failt erst bei der Abarbeitung von Requests. Da ist das Feedback für den Konfigurierenden viel zu spät.

Reply:
es ist ja durchaus valide das feature zu deaktivieren. Es ist nur nicht valide dann trotzdem den service zu rufen.

Wir wissen nicht dass es ein fehler ist bis ein request bearbeitet werden soll.